File: olympic10-hcm/2016-2017/doxingau/doxingau.py
import os
import sys

input_file = os.path.join(sys.path[0], 'doxingau.inp')
output_file = os.path.join(sys.path[0], 'doxingau.out')

s1 = s2 = s3 = 0

# mảng tần số lưu số lần xuất hiện của mỗi tổng
counter = 0
result = 0


def input():
    global s1, s2, s3

    sys.stdin = open(input_file, 'r')

    # Đọc toàn bộ dữ liệu vào bộ nhớ đệm và tách thành các token
    data = sys.stdin.read().split()

    iterator = iter(data)

    s1 = int(next(iterator))
    s2 = int(next(iterator))
    s3 = int(next(iterator))


def process():
    global result

    # Khởi tạo mảng tần số
    counter = [0] * (s1 + s2 + s3 + 1)

    # Duyệt từng mặt của từng xí ngầu
    for i in range(1, s1 + 1):
        for j in range(1, s2 + 1):
            for k in range(1, s3 + 1):
                # Tăng số lần xuất hiện của tổng thêm 1
                counter[i + j + k] += 1

    # Lấy chỉ số của phần tử lớn nhất, cũng chính là tổng của 3 xí ngầu
    result = counter.index(max(counter))


def output():
    sys.stdout = open(output_file, 'w')

    print(result)


if __name__ == '__main__':
    input()
    process()
    output()

File: olympic10-hcm/2016-2017/doxingau/doxingau.cpp
#include <iostream>
#include <vector>
#include <algorithm>

#define input_file "doxingau.inp"
#define output_file "doxingau.out"

using namespace std;

int s1, s2, s3;
vector<int> counter;
int result;

void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    freopen(input_file, "r", stdin);

    cin >> s1 >> s2 >> s3;
}

void process()
{
    // Khởi tạo mảng tần số
    counter.resize(s1 + s2 + s3 + 1, 0);

    // Duyệt từng mặt của từng xí ngầu
    for (int i = 1; i <= s1; ++i)
    {
        for (int j = 1; j <= s2; ++j)
        {
            for (int k = 1; k <= s3; ++k)
            {
                // Tăng số lần xuất hiện của tổng thêm 1
                counter[i + j + k]++;
            }
        }
    }

    // Tìm vị trí của phần tử lớn nhất
    auto it = max_element(counter.begin(), counter.end());

    // Lấy chỉ số của phần tử lớn nhất, cũng chính là tổng của 3 xí ngầu
    result = distance(counter.begin(), it);
}

void output()
{
    freopen(output_file, "w", stdout);

    cout << result;
}

int main()
{
    input();
    process();
    output();

    return 0;
}